gui/window.cpp

```diff
@@ -0,0 +1,587 @@
+    Cluster clusterHash = getClusterHash()[cluster];
+    if (clusterHash.status() == "Running") {
+        startButton->setEnabled(false);
+        stopButton->setEnabled(true);
+#if __linux__
+        sshButton->setEnabled(true);
+#endif
+        dashboardButton->setEnabled(true);
+    } else {
+        startButton->setEnabled(true);
+        stopButton->setEnabled(false);
+    }
+}
+
+ClusterHash Window::getClusterHash()
+{
+    ClusterList clusters = getClusters();
+    ClusterHash clusterHash;
+    for (int i = 0; i < clusters.size(); i++) {
+        Cluster cluster = clusters.at(i);
+        clusterHash[cluster.name()] = cluster;
+    }
+    return clusterHash;
+}
+
+bool Window::sendMinikubeCommand(QStringList cmds)
+{
+    QString text;
+    return sendMinikubeCommand(cmds, text);
+}
+
+bool Window::sendMinikubeCommand(QStringList cmds, QString &text)
+{
+    QString program = minikubePath();
+    if (program.isEmpty()) {
+        return false;
+    }
+    QStringList arguments;
+    arguments << cmds;
+    bool success;
+
+    QProcess *process = new QProcess(this);
+    process->start(program, arguments);
+    this->setCursor(Qt::WaitCursor);
+    success = process->waitForFinished(300 * 1000);
+    this->unsetCursor();
+
+    if (success) {
+        text = process->readAllStandardOutput();
+    } else {
+        qDebug() << process->readAllStandardOutput();
+        qDebug() << process->readAllStandardError();
+    }
+    delete process;
+    return success;
+}
+
+static QString profile = "minikube";
+static int cpus = 2;
+static int memory = 2400;
+static QString driver = "";
+static QString containerRuntime = "";
+
+void Window::askName()
+{
+    QDialog dialog;
+    dialog.setWindowTitle(tr("Create minikube Cluster"));
+    dialog.setModal(true);
+
+    QFormLayout form(&dialog);
+    QDialogButtonBox buttonBox(Qt::Horizontal, &dialog);
+    QLineEdit profileField(profile, &dialog);
+    form.addRow(new QLabel(tr("Profile")), &profileField);
+    buttonBox.addButton(QString(tr("Use Default Values")), QDialogButtonBox::AcceptRole);
+    connect(&buttonBox, &QDialogButtonBox::accepted, &dialog, &QDialog::accept);
+    buttonBox.addButton(QString(tr("Set Custom Values")), QDialogButtonBox::RejectRole);
+    connect(&buttonBox, &QDialogButtonBox::rejected, &dialog, &QDialog::reject);
+    form.addRow(&buttonBox);
+
+    int code = dialog.exec();
+    profile = profileField.text();
+    if (code == QDialog::Accepted) {
+        QStringList arg = { "start", "-p", profile };
+        sendMinikubeCommand(arg);
+    } else if (code == QDialog::Rejected) {
+        askCustom();
+    }
+}
+
+void Window::askCustom()
+{
+    QDialog dialog;
+    dialog.setWindowTitle(tr("Set Cluster Values"));
+    dialog.setModal(true);
+
+    QFormLayout form(&dialog);
+    driverComboBox = new QComboBox;
+    driverComboBox->addItem("docker");
+    driverComboBox->addItem("hyperv");
+    driverComboBox->addItem("hyperkit");
```

Comment:
the "kvm2" driver is missing in this list, for Linux users.

Also "virtualbox" should probably be available, no ?

Reply:
Added all the drivers and gated OS specific ones